docs/docs/server-side-apply.md

```diff
@@ -0,0 +1,36 @@
+---
+name: "Server-Side Apply"
+route: "/server-side-applye"
+menu: Advanced features
+---
+
+# Server-Side Apply
+
+Tanka supports [server-side apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/),
+which requires at least Kubernetes 1.16+, and was promoted to stable status in 1.22.
+
+To enable server-side diff in tanka, add the following field to `spec.json`:
+
+```diff
+{
+  "spec": {
++    "applyStrategy": "server",
+  }
+}
+```
+
+While server-side apply doesn't have any effect on the resources being applied
+and is intended to be a general in-place upgrade to client-side apply, there are
+differences in how fields are managed that can make converting existing cluster
+resources a non-trival change.
+
+Identifying and fixing these changes are beyond the scope of this guide, but
+many can be found before an apply with the `validate`
+[diff strategy](/diff-strategy).
```

Comment:
This PR already looks great.

It may be a little out-of-scope but I was wondering whether we should make this consistent for both diff and apply. 
`tk diff` could also honor the `applyStrategy` option if set in spec.json and it could have a CLI flag (replacing diff-strategy=validate): `tk diff --diff-strategy=native --server-side` (native is the default strategy)

@julienduchesne do you think this makes sense?

Reply:
I think `--diff-strategy=validate` should be renamed (or aliased) to `--diff-strategy=server`, that would be consistent enough IMO. Having both a `--server-side` and a `--(diff|apply)-strategy` might get confusing. 

Also, the `diffStrategy` and `applyStrategy` settings are nice since you can toggle them independently in the spec (someone may wish to apply server-side but do a simple diff, or vice-versa, server-side sometimes has a weird behavior)